.devcontainer/Dockerfile

```diff
@@ -35,15 +37,22 @@ RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
 RUN update-ca-certificates
 
 # Terraform
-RUN curl -fsSL https://apt.releases.hashicorp.com/gpg | sudo apt-key add - \
-	&& apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main" \
-	&& apt-get update && sudo apt-get -qq install --no-install-recommends --yes terraform terraform-ls
+# hashicorp only provides amd64 packages, so we have to make do
+ARG TERRAFORM_VERSION=1.3.6
+RUN mkdir -p /tmp/download \
+	&& cd /tmp/download \
+	&& wget -O terraform.zip "https://releases.hashicorp.com/terraform/1.3.6/terraform_1.3.6_linux_${TARGETARCH}.zip" \
+	&& unzip -qq ./terraform.zip \
+	&& mv ./terraform /usr/local/bin/terraform \
+	&& rm -rf /tmp/download
 
 # AWS CLI
 SHELL ["/bin/zsh", "-c"]
 RUN mkdir -p /tmp/download \
 	&& cd /tmp/download \
-	&& curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" --silent -o "awscliv2.zip" \
+	&& ARCH=${TARGETARCH/amd64/x86_64} \
```

Comment:
Why provide `ARCH` twice here?

Reply:
because of the awscli hosting URLs, this needs to replace amd64 => x86_64 and arm64 => aarch64. Since I didn't want to touch the TARGETARCH, this code is overwriting ARCH a second time to apply both replaces.